### PR TITLE
Support to run additional commands

### LINKED ISSUE #100: Add monitoring and reporting for Jenkins SSL certs expiration.
### Is your feature request related to a problem? Please describe

Currently there is no monitoring setup for emitting DaysToExpiry metric for SSL Certs being used for Jenkins setup. This is critical security gap as expired certificate will risk compromised encryption and authentication. This is related to meta issue #

### Describe the solution you'd like

The plan is to pass the ssl cert DaysToExpiry  metric script as an additional context argument to the cdk app and run it as cron on the jenkins master ec2 node. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rishabh6788**
Signed-off-by: Rishabh Singh <sngri@amazon.com>

### Description
This PR adds support to run custom scripts or commands on any of the desired ec2 instances. If someone is looking to run additional logic, such as a script that pushes SSL Cert expiry metric to desired monitoring portal every 5-min, they can pass the script's path.

To utilise this functionality the user will have to pass optional context argument `additionalCommands` - takes the path to the script as string.

### Issues Resolved
closes #100 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/compute/run-additional-commands.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      runCommand.push('echo "*/5 * * * * $HOME/emitter " > newcron');
+    } else {
+      runCommand.push(`echo "*/5 * * * * $HOME/emitter ${additionalParams.toString()}" > newcron`);
+    }
+    runCommand.push('crontab newcron');
+    runCommand.push('rm newcron');
+
+    return runCommand;
+  }
+}
```
Comment (gaiksaya) [inline lib/compute/run-additional-commands.ts:64]: I don't think this concerns the external user. Can we take this as args (array of strings) and then push one by one?

File: lib/ci-stack.ts [hunk 2]
```diff
@@ -54,18 +59,22 @@ export class CIStack extends Stack {
...
-      throw new Error('useSsl parameter is required to be set as - true or false');
+      throw new Error('useSsl parameter is required to be set as - true or false. Please pass the value');
     }
 
     const useSsl = useSslParameter === 'true';
 
     const runWithOidcParameter = `${props?.runWithOidc ?? this.node.tryGetContext('runWithOidc')}`;
     if (runWithOidcParameter !== 'true' && runWithOidcParameter !== 'false') {
-      throw new Error('runWithOidc parameter is required to be set as - true or false');
+      throw new Error('runWithOidc parameter is required to be set as - true or false. Pass the damn value');
```
Comment (gaiksaya) [inline lib/ci-stack.ts:69]: Please change to `Please pass the value` if at all required

**Review by gaiksaya — CHANGES_REQUESTED** [review]

File: lib/compute/jenkins-main-node.ts [hunk 3]
```diff
@@ -199,6 +200,7 @@ export class JenkinsMainNode {
       InitPackage.yum('mod_ssl'),
       // eslint-disable-next-line max-len
       InitCommand.shellCommand('sudo wget -nv https://github.com/mikefarah/yq/releases/download/v4.22.1/yq_linux_amd64 -O /usr/bin/yq && sudo chmod +x /usr/bin/yq'),
+      InitCommand.shellCommand('python3 -m pip install --upgrade pip && python3 -m pip install cryptography boto3 requests-aws4auth'),
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply lib/compute/jenkins-main-node.ts:234]: Yes, it is failing for boto3 installation as it is trying to use python2.7 to install boto3 library version which is no longer supports python2.7.

File: lib/compute/run-additional-commands.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import { Stack } from '@aws-cdk/core';
+import { CfnAssociation, CfnAssociationProps, CfnDocument } from '@aws-cdk/aws-ssm';
+
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply lib/compute/run-additional-commands.ts]: Updated.

File: lib/compute/run-additional-commands.ts (hunk 1 above)
Comment (rishabh6788) [reply lib/compute/run-additional-commands.ts]: agreed. Updated the logic to send all the required commands from command line.

File: lib/ci-stack.ts [hunk 5]
```diff
@@ -116,5 +125,11 @@ export class CIStack extends Stack {
     });
 
     const monitoring = new JenkinsMonitoring(this, externalLoadBalancer, mainJenkinsNode);
+
+    if (additionalCommandsContext.toString() === 'undefined') {
+      console.log('No additional commands passed. Nothing to do.');
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply lib/ci-stack.ts]: removed.

File: lib/ci-stack.ts (hunk 2 above)
Comment (rishabh6788) [reply lib/ci-stack.ts]: Updated.

File: resources/hello-world.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+"""
```
[the remark this answers is not in the document]
Comment (rishabh6788) [reply resources/hello-world.py]: Done.

**Comment by gaiksaya** [comment]
Keeping this PR on hold until #106 is resolved.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**